repose-aggregator/components/filters/valkyrie-authorization/src/main/scala/org/openrepose/filters/valkyrieauthorization/ValkyrieAuthorizationFilter.scala

```diff
@@ -233,25 +233,25 @@ class ValkyrieAuthorizationFilter @Inject()(configurationService: ConfigurationS
     } else {
       val checkHeader = checkHeaders(requestedTenantId, requestedContactId)
       val userPermissions = getPermissions(checkHeader)
-      val authPermissions = authorizeDevice(userPermissions, requestedDeviceId)
-      val allPermissions =  authPermissions match {
+      val allPermissions = userPermissions match {
         case UserPermissions(deviceRoles, devicePermissions) =>
           if (!configuration.isEnableBypassAccountAdmin && deviceRoles.contains(ACCOUNT_ADMIN)) {
               getInventory(checkHeader) match {
               case UserPermissions(adminRoles, adminPermissions) =>
                 UserPermissions(deviceRoles ++ adminRoles, devicePermissions ++ adminPermissions)
-              case _ => authPermissions
+              case _ => userPermissions
```

Comment:
This is the wrong thing to return here. You've effectively swallowed if there were any errors with the second valkyrie call. This should instead return the output of getInventory.